source/navier_stokes_projection/diffusion_step_methods.cc

```diff
@@ -72,26 +81,32 @@ solve_diffusion_step(const bool reinit_prec)
     build_preconditioner(diffusion_step_preconditioner,
                          *system_matrix,
                          solver_parameters.preconditioner_parameters_ptr,
-                         (velocity->fe_degree > 1? true: false));
+                         (velocity->fe_degree > 1? true: false),
+                         symmetric_system_matrix);
   }
 
   AssertThrow(diffusion_step_preconditioner != nullptr,
               ExcMessage("The pointer to the diffusion step's preconditioner has not being initialized."));
 
+  const double rhs_norm{diffusion_step_rhs.l2_norm()};
+
   SolverControl solver_control(
     parameters.diffusion_step_solver_parameters.n_maximum_iterations,
-    std::max(solver_parameters.relative_tolerance * diffusion_step_rhs.l2_norm(),
+    std::max(solver_parameters.relative_tolerance * rhs_norm,
              solver_parameters.absolute_tolerance));
 
   #ifdef USE_PETSC_LA
     LinearAlgebra::SolverGMRES solver(solver_control,
-                                      MPI_COMM_WORLD);
+                                      mpi_communicator);
   #else
     LinearAlgebra::SolverGMRES solver(solver_control);
   #endif
 
   try
   {
+    if (rhs_norm < std::numeric_limits<float>::epsilon())
+      distributed_velocity = 1.0;
```

Comment:
What is the reasoning behind this? A better initial guess?

Reply:
The reason is a very simple bug fix: In the MIT benchmark, the residual is exactly equal to zero in the first time step. In this case, the linear solver somehow does not converge. Setting all entries of the solution vector to zero yields a residual `1e-6` which makes the solver converge.

Reply:
> The reason is a very simple bug fix: In the MIT benchmark, the residual is exactly equal to zero in the first time step.

Just to be clear, you mean the residual of the solver?

> In this case, the linear solver somehow does not converge

Did you try playing with the tolerance values of the first step? Or the preconditioner settings?

> Setting all entries of the solution vector to zero yields a residual `1e-6` which makes the solver converge. 

To zero or to one?

Reply:
> Just to be clear, you mean the residual of the solver?

I mean || A * x - b||.

> Did you try playing with the tolerance values of the first step? Or the preconditioner settings?

Yes, it did not help...

> To zero or to one?

To one of course.